src/types/type.ts

```diff
@@ -89,7 +89,9 @@ export abstract class ComplexType<S, T> implements IType<S, T> {
     }
 
     reconcile(current: Node, newValue: any): Node {
-        const { parent, subpath } = current
+        if (current.snapshot === newValue)
```

Comment:
Q: is this check performed _before_ apply preProcessedSnapshot hooks? I think it should be before, for symmetry purposes, as the stored snapshot is the postProcessedSnapshot. Not sure about it, but there might be problems if this optimization is applied after preProcessing (although I can't think of one, maybe you can :-P).

Further LGTM :)